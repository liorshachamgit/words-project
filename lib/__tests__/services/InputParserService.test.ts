const InputService = require('../../modules/services/InputParserService')

let service

beforeAll(() => {
    service = new InputService.InputParserService()
})

describe('Test ParseInput validation', () => {
    it('should fail when input is null', async () => {
        try {
            const res = await service.ParseInput(null)
        } catch (e) {
            expect(e).not.toBeNull()
            expect(e).toEqual('bad input')
        }
    })

    it('should fail when input is undefined', async () => {
        try {
            const res = await service.ParseInput(undefined)
        } catch (e) {
            expect(e).not.toBeNull()
            expect(e).toEqual('bad input')
        }
    })

    it('should fail when input is undefined', async () => {
        try {
            const res = await service.ParseInput()
        } catch (e) {
            expect(e).not.toBeNull()
            expect(e).toEqual('bad input')
        }
    })

    it('should fail when input is not a string', async () => {
        try {
            const res = await service.ParseInput({})
        } catch (e) {
            expect(e).not.toBeNull()
            expect(e).toEqual('bad input')
        }
    })
})

describe('Regular text input - ', () => {
    it('should return object with occurences count when receving the same word', async () => {
        const input = `hello, hello HELlo HELLO , , hellO`
        const res = await service.ParseInput(input)
        expect(res).not.toBeNull()
        expect(res.hello).toEqual(5)
    })

    it('should return object with occurences count', async () => {
        const input = `I'm Slim Shady, yes I'm the real Shady
                        All you other Slim Shadys, are just imitating
                        So won't the real Shady, please stand up
                        Please stand up, please stand up`
        const res = await service.ParseInput(input)
        expect(res).not.toBeNull()
        expect(res.slim).toEqual(2)
        expect(res.shady).toEqual(3)
        expect(res).toMatchSnapshot()

    })

})

describe('File system input - ', () => {
    describe('Failing inputs', () => {
        it('should return error when directory not found', async () => {
            try {
                await service.ParseInput('c:\\non-existing-dir\\file.txt')
            } catch (e) {
                expect(e).toContain('no such file or directory')
            }
        });

        it('should return error when file not found', async () => {
            try {
                await service.ParseInput('C:\\Users\\lior\\Documents\\Projects\\word project\\test files\\missingFile.txt')
            } catch (e) {
                expect(e).toMatch('no such file or directory')
            }
        });

        // it.each([
        //     'C:\\Users\\lior\\Documents\\Projects\\word project\\test files\\input.jpg',
        //     'C:\\Users\\lior\\Documents\\Projects\\word project\\test files\\email.csv',
        //     'C:\\Users\\lior\\Documents\\Projects\\word project\\test files\\input'
        //     ])('should return error when file has unsupported extension', async (path) => {
        //     try {
        //         await service.ParseInput(path)
        //     } catch (e) {
        //         expect(e).toMatch('Unsupported file extension. The only supported files are from .txt')
        //     }
        // });

    });

    describe('Success inputs', () => {
        it('should return occurences object for full path', async () => {
            const data = await service.ParseInput('C:\\Users\\lior\\Documents\\Projects\\word project\\test files\\forgetAboutDre.txt')
            expect(data).not.toBeNull()
            expect(data.you).toEqual(26)
            expect(data.dre).toEqual(10)
            expect(data).toMatchSnapshot()
        });

        it('should return occurences object for relative path', async () => {
            const data = await service.ParseInput('./test files/Smells Like Teen Spirit.txt')
            expect(data).not.toBeNull()
            expect(data.hello).toEqual(36)
            expect(data.entertain).toEqual(6)
            expect(data).toMatchSnapshot()
        });
    });
});

describe('URL path input - ', () => {
    it('should get content of https request ', async () => {
        const url = 'https://baconipsum.com/api/?type=meat-and-filler&paras=100&format=text'
        const data = await service.ParseInput(url)
        expect(data).not.toBeNull()
    });

    // it.each(['json', 'html'])('should fail when trying to read format that is not text ', async (format) => {
    //     try {
    //         const url = `https://baconipsum.com/api/?type=meat-and-filler&paras=100&format=${format}`
    //         await service.ParseInput(url)
    //     } catch (e) {
    //         expect(e).not.toBeNull()
    //     }
    // });
});