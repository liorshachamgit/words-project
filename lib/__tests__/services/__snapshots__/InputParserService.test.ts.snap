// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File system input -  Success inputs should return occurences object for full path 1`] = `
Object {
  "80s": 1,
  "a": 29,
  "about": 8,
  "act": 7,
  "add": 1,
  "afford": 1,
  "after": 1,
  "aftermath": 1,
  "aint": 4,
  "album": 1,
  "all": 11,
  "an": 1,
  "and": 26,
  "anybody": 1,
  "are": 2,
  "around": 2,
  "as": 1,
  "ass": 1,
  "at": 2,
  "awkward": 1,
  "babies": 1,
  "back": 3,
  "barkin": 1,
  "be": 1,
  "beats": 1,
  "been": 4,
  "benz": 1,
  "better": 2,
  "bitch": 1,
  "bloodier": 1,
  "blow": 1,
  "boats": 1,
  "both": 1,
  "bow": 1,
  "bring": 1,
  "broken": 1,
  "brought": 1,
  "bump": 1,
  "bumpin": 1,
  "buncha": 7,
  "burnt": 1,
  "but": 14,
  "by": 2,
  "cabbage": 1,
  "call": 1,
  "callin": 1,
  "came": 1,
  "can": 4,
  "car": 1,
  "cats": 1,
  "caught": 1,
  "cause": 4,
  "cd": 1,
  "charleston": 1,
  "cheese": 1,
  "chew": 1,
  "choke": 1,
  "chronic": 1,
  "closely": 1,
  "come": 1,
  "comes": 7,
  "comin": 1,
  "company": 1,
  "compton": 1,
  "cops": 2,
  "crazy": 1,
  "crib": 1,
  "crumb": 1,
  "cubes": 1,
  "cuddled": 1,
  "d": 1,
  "damn": 1,
  "dark": 1,
  "day": 2,
  "deals": 1,
  "death": 1,
  "disease": 1,
  "do": 1,
  "doc": 2,
  "docs": 1,
  "dodge": 1,
  "dogs": 1,
  "doin": 1,
  "dont": 2,
  "dope": 1,
  "double": 1,
  "dough": 1,
  "down": 2,
  "dre": 10,
  "drunk": 1,
  "eazyes": 1,
  "em": 3,
  "enough": 1,
  "everybody": 8,
  "eye": 1,
  "family": 1,
  "fell": 1,
  "finally": 1,
  "firm": 1,
  "flop": 1,
  "forgot": 7,
  "found": 1,
  "free": 1,
  "freeze": 1,
  "from": 3,
  "fuck": 5,
  "fuckin": 1,
  "full": 5,
  "g": 1,
  "gangstas": 1,
  "garage": 1,
  "gas": 1,
  "gats": 1,
  "gave": 1,
  "get": 3,
  "getting": 1,
  "gibberish": 7,
  "give": 2,
  "givin": 1,
  "go": 2,
  "goes": 1,
  "gonna": 3,
  "good": 1,
  "got": 9,
  "gots": 1,
  "groceries": 1,
  "group": 1,
  "gs": 2,
  "guns": 1,
  "guy": 1,
  "habitat": 1,
  "hailey": 1,
  "hand": 1,
  "hands": 1,
  "hangin": 1,
  "harder": 1,
  "hate": 2,
  "hated": 1,
  "have": 1,
  "havin": 1,
  "he": 1,
  "helped": 1,
  "here": 1,
  "him": 2,
  "his": 1,
  "hood": 1,
  "hoppin": 1,
  "hotter": 1,
  "house": 2,
  "humungous": 1,
  "i": 15,
  "ice": 1,
  "if": 3,
  "im": 6,
  "in": 10,
  "is": 3,
  "it": 5,
  "its": 5,
  "ive": 1,
  "jungle": 1,
  "just": 8,
  "kani": 1,
  "karl": 1,
  "keep": 1,
  "key": 1,
  "keys": 1,
  "kill": 1,
  "knees": 1,
  "knew": 1,
  "know": 1,
  "lab": 1,
  "label": 1,
  "ladies": 1,
  "last": 1,
  "legs": 1,
  "let": 1,
  "like": 20,
  "lips": 7,
  "listen": 1,
  "little": 1,
  "loco": 1,
  "lookin": 1,
  "lot": 1,
  "loud": 1,
  "low": 1,
  "mad": 2,
  "mail": 1,
  "matches": 1,
  "me": 18,
  "men": 1,
  "mercedes": 1,
  "mid": 1,
  "millenium": 1,
  "more": 1,
  "most": 1,
  "mould": 1,
  "move": 7,
  "muthafuck": 1,
  "muthafuckas": 8,
  "muthafuckin": 1,
  "my": 4,
  "new": 2,
  "next": 2,
  "niggas": 2,
  "no": 10,
  "none": 1,
  "not": 1,
  "nothin": 8,
  "now": 3,
  "nowadays": 7,
  "nwa": 1,
  "o": 1,
  "of": 11,
  "off": 5,
  "office": 1,
  "okay": 1,
  "ol": 1,
  "old": 1,
  "oldies": 1,
  "on": 4,
  "one": 2,
  "or": 3,
  "out": 12,
  "pad": 1,
  "park": 1,
  "parkin": 1,
  "patch": 1,
  "pen": 1,
  "plaque": 1,
  "plaques": 1,
  "platinum": 1,
  "please": 1,
  "police": 1,
  "pop": 1,
  "provide": 1,
  "rap": 1,
  "rappers": 1,
  "reason": 1,
  "resolve": 1,
  "right": 1,
  "run": 1,
  "said": 2,
  "sales": 1,
  "same": 1,
  "savage": 1,
  "save": 1,
  "say": 8,
  "sayin": 1,
  "see": 1,
  "set": 1,
  "shady": 1,
  "skis": 1,
  "sleep": 1,
  "slim": 1,
  "smoke": 1,
  "snoop": 1,
  "snowmobiles": 1,
  "so": 4,
  "sold": 1,
  "some": 1,
  "somebody": 1,
  "somethin": 1,
  "something": 7,
  "sorry": 1,
  "startin": 1,
  "stay": 1,
  "still": 3,
  "stood": 1,
  "stop": 1,
  "strangled": 1,
  "strapped": 1,
  "streets": 1,
  "stroll": 1,
  "studio": 1,
  "study": 1,
  "talk": 8,
  "tape": 2,
  "taught": 1,
  "temp": 1,
  "than": 1,
  "that": 9,
  "the": 23,
  "them": 2,
  "then": 1,
  "there": 1,
  "these": 1,
  "they": 29,
  "things": 1,
  "think": 5,
  "this": 3,
  "through": 2,
  "to": 24,
  "today": 1,
  "told": 2,
  "tomorrows": 1,
  "too": 2,
  "tracks": 1,
  "trees": 1,
  "trophies": 1,
  "trouble": 1,
  "truck": 1,
  "tryna": 4,
  "turn": 1,
  "turned": 1,
  "twin": 1,
  "two": 3,
  "up": 7,
  "walk": 1,
  "walkin": 1,
  "walkman": 1,
  "wall": 1,
  "wanna": 10,
  "want": 1,
  "was": 5,
  "wasnt": 2,
  "way": 3,
  "well": 1,
  "were": 1,
  "what": 3,
  "wheels": 1,
  "when": 13,
  "wheres": 1,
  "who": 3,
  "whos": 1,
  "windows": 1,
  "wit": 13,
  "with": 1,
  "wont": 1,
  "would": 1,
  "wouldnt": 1,
  "yall": 8,
  "you": 26,
  "your": 4,
}
`;

exports[`File system input -  Success inputs should return occurences object for relative path 1`] = `
Object {
  "a": 16,
  "albino": 3,
  "always": 2,
  "an": 3,
  "and": 8,
  "are": 6,
  "at": 1,
  "been": 1,
  "best": 1,
  "blessed": 1,
  "bring": 1,
  "contagious": 3,
  "dangerous": 3,
  "denial": 9,
  "dirty": 1,
  "do": 1,
  "end": 1,
  "entertain": 6,
  "feel": 4,
  "find": 1,
  "for": 1,
  "forget": 1,
  "found": 1,
  "friends": 1,
  "fun": 1,
  "gift": 1,
  "group": 1,
  "guess": 1,
  "guns": 1,
  "hard": 2,
  "has": 1,
  "hello": 36,
  "here": 6,
  "hey": 2,
  "how": 9,
  "i": 10,
  "im": 1,
  "it": 2,
  "its": 5,
  "just": 1,
  "know": 1,
  "less": 3,
  "libido": 3,
  "lights": 3,
  "little": 1,
  "load": 1,
  "lose": 1,
  "low": 9,
  "makes": 1,
  "me": 1,
  "mind": 1,
  "mosquito": 3,
  "mulatto": 3,
  "my": 3,
  "never": 1,
  "no": 1,
  "now": 6,
  "oh": 3,
  "on": 1,
  "our": 1,
  "out": 3,
  "overbored": 1,
  "pretend": 1,
  "selfassured": 1,
  "shes": 1,
  "smile": 1,
  "stupid": 3,
  "taste": 1,
  "the": 4,
  "this": 1,
  "to": 3,
  "until": 1,
  "up": 1,
  "us": 6,
  "we": 6,
  "well": 1,
  "what": 1,
  "whatever": 1,
  "why": 1,
  "will": 1,
  "with": 3,
  "word": 1,
  "worse": 1,
  "yeah": 3,
  "your": 1,
}
`;

exports[`Regular text input -  should return object with occurences count 1`] = `
Object {
  "all": 1,
  "are": 1,
  "im": 2,
  "imitating": 1,
  "just": 1,
  "other": 1,
  "please": 3,
  "real": 2,
  "shady": 3,
  "shadys": 1,
  "slim": 2,
  "so": 1,
  "stand": 3,
  "the": 2,
  "up": 3,
  "wont": 1,
  "yes": 1,
  "you": 1,
}
`;
